lib/http/http.go

```diff
@@ -353,6 +354,7 @@ func start() error {
 	}
 	defaultServer = s.(*server)
 	defaultServer.Serve()
+	defaultServer.Router().Use(middleware.Compress(5, CompressibleTypes...))
```

Comment:
Use better-than-stdlib compressors:

```suggestion
	comp := middleware.Compress(5, CompressibleTypes...)
	comp.SetEncoder("gzip", func(w io.Writer, level int) io.Writer {
		gw, err := gzip.NewWriterLevel(w, level)
		if err != nil {
			return nil
		}
		return gw
	})
	comp.SetEncoder("deflate", func(w io.Writer, level int) io.Writer {
		dw, err := flate.NewWriter(w, level)
		if err != nil {
			return nil
		}
		return dw
	})
	defaultServer.Router().Use(comp)
```

Import
```
	"github.com/klauspost/compress/flate"
	"github.com/klauspost/compress/gzip"
```

Deflate is typically not used, but it seems like this middleware uses it, so maybe it is ok.

Reply:
I am not sure what this accomplishes over the default implementation:

https://github.com/go-chi/chi/blob/c9e87efe9691a63d6a89de8bbd16b04fe4d6640e/middleware/compress.go#L94-L123

Can you clarify a bit? The encoder used is specified by the client, so long as the server supports it.

Reply:
@innovate-invent You replace the compressors with the other imports. They are faster ,typically 1.8x the throughput on web data.

Reply:
Understood